src/BuildIt.Forms/BuildIt.Forms.Controls/Platforms/Android/CameraPreviewControlRenderer.cs

```diff
@@ -379,24 +448,38 @@ internal void OpenCamera(int width, int height)
             CameraDevice = null;
             SetUpCameraOutputs(width, height);
             ConfigureTransform(width, height);
+            var errorOccurred = false;
             var manager = (CameraManager)Context.GetSystemService(Context.CameraService);
             try
             {
                 if (!CameraOpenCloseLock.TryAcquire(2500, TimeUnit.Milliseconds))
                 {
-                    throw new RuntimeException("Time out waiting to lock camera opening.");
+                    cameraPreviewControl.RaiseErrorOpeningCamera();
+                    return;
                 }
 
                 manager.OpenCamera(cameraId, stateCallback, BackgroundHandler);
             }
             catch (CameraAccessException e)
             {
+                errorOccurred = true;
                 e.PrintStackTrace();
             }
             catch (InterruptedException e)
             {
+                errorOccurred = true;
                 throw new RuntimeException("Interrupted while trying to lock camera opening.", e);
```

Comment:
We're still throwing the runtime exception? Isn't the intent to prevent crashing the app by providing an alternative way for the error to be handled

Reply:
yep missed that. will change